### ISSUE TITLE
[FEAT]:  LDAP server

### ISSUE THREAD

**Issue by traverseda** [azukaar/Cosmos-Server#249]
### Feature Description

There are some services that only support LDAP. This will probably always be true. One that really affects end users is jellyfin. For more corporate users all the git servers, gitea, bitbucket, etc, all work much better when using LDAP directly.

https://github.com/glauth/glauth is a nice go library for implementing LDAP servers.

**Comment by zidesm** [comment]
Please add LDAP support for the users.

Maybe in the future roles could be assigned to specify why users or group can access certain applications.

**Comment by lyrameg** [comment]
Jellyfin has a plugin which can be installed. (https://github.com/9p4/jellyfin-plugin-sso/tree/main)
create an entry in cosmos openID and paste that into "OpenID Client ID" and "OID Secret" in the SSO config
then you have to fill out "OID Endpoint" with `https://YOUR_DOMAIN/.well-known/openid-configuration`
finally check the box "Enabled", enable the folders you want them to access(or check the box to enable all) and click Save.

Though it would be nice to have support for LDAP for other apps, but you can use Jellyfin with openID

**Comment by gianlazz** [comment]
@lyrameg correct me if I'm wrong but I think the issue with jellyfin and OIDC is with signing in via 3rd party apps.

**Comment by traverseda** [comment]
Yeah, I don't think that OIDC plugin will work with my android TV client.

You also need LDAP for things like gitea, since git clients don't support OIDC. It's impractical to use OIDC to authenticate an SSH session for example.

**Comment by azukaar** [comment]
> Yeah, I don't think that OIDC plugin will work with my android TV client.
> 
> You also need LDAP for things like gitea, since git clients don't support OIDC. It's impractical to use OIDC to authenticate an SSH session for example.

WDYM dont you just create an SSH token on your account regardless of your web connection method

**Comment by traverseda** [comment]
No, that's some weird bullshit that github invented. I don't want to create a new token for every computer I deploy my dotfiles on. Git has always worked with password authentication.

**Comment by azukaar** [comment]
I think you're being a little mislead, the base git client does support SSH authentication natively, regardless of github, and it is also the best way to secure your account, because that SSH token only allows using git actions, and cannot authenticate to the web admin, also it is well kept by your system and harder to compromise.
Additionally deploying a key per device is normal for SSL.
And FINALLY not only the base git client supports SSH keys auth just fine, github also allows password authentication regarless (both from the git client and their git cli). 
I think you're getting yourself worked up on something that simply isn't quite as bad as you think
